ros_buildfarm/templates/snippet/publisher_warnings.xml.em

```diff
@@ -14,6 +14,13 @@
           <reportEncoding></reportEncoding>
           <skipSymbolicLinks>false</skipSymbolicLinks>
         </io.jenkins.plugins.analysis.warnings.Gcc4>
+        <io.jenkins.plugins.analysis.warnings.ClangTidy>
+          <id></id>
+          <name></name>
+          <pattern>@[if build_tool == 'colcon']ws/log/test_*/*/stdout_stderr.log@[end if]</pattern>
```

Comment:
Scanning test logs rather than build logs is intentional here since ament_lint_clang_tidy is run as part of the colcon test phase.

Reply:
Doesn't stdout/stderr end up getting forwarded to the console? Does this plugin **require** a file instead?

Reply:
Also, if we have to use colcon logs, can we use the `latest_test` symlink instead of `test_*`?

Reply:
> Doesn't stdout/stderr end up getting forwarded to the console? Does this plugin require a file instead?


I was following the convention of the existing warning publishers by targeting the log files.

I think I like the idea of scanning the log files rather than trusting that the console output will remain unchanged but I'm pretty sure that idea only holds if I also think it's more likely that we'd change the event handlers for colcon than that we'd change the workspace or test log location. And I don't know that I actually think that...

> Also, if we have to use colcon logs, can we use the latest_test symlink instead of test_*?

I don't have any objection to that either, again I was just matching the convention set by the existing warnings publishers.

I've triggered a test build with no file pattern thus just scanning the console log to see if there is any difference in results that would motivate me to keep the file pattern.

Reply:
The results are the same scanning the console log.

Reply:
93301f0 removes the pattern, running the warning parser on the build log rather than on any files in the job workspace.